src/ui/components/NetworkMonitor/ResponseBody.tsx

```diff
@@ -0,0 +1,54 @@
+import { ResponseBodyData } from "@recordreplay/protocol";
+import React, { useState } from "react";
+import ReactJson from "react-json-view";
+import { TriangleToggle } from "./RequestDetails";
+import { RequestSummary } from "./utils";
+
+const FIVE_MEGABYTES = 5e6;
+
+const safeJSONParseBase64 = (maybeJSON: string) => {
+  try {
+    return [true, JSON.parse(atob(maybeJSON))];
+  } catch (e) {
+    console.log(maybeJSON);
+    console.log(e);
+    return [false, maybeJSON];
+  }
+};
+
+const ResponseBody = ({
+  request,
+  responseBodyParts,
+}: {
+  request?: RequestSummary;
+  responseBodyParts: ResponseBodyData[];
+}) => {
+  if (!request || !responseBodyParts?.length) {
+    return null;
+  }
+
+  const contentLength = request.requestHeaders.find(h => h.name.toLowerCase() === "content-length");
+  const [validJSON, content] = safeJSONParseBase64(responseBodyParts.map(p => p.value).join(""));
+  const [expanded, setExpanded] = useState(true);
+
+  return (
+    <div className="bg-white w-full overflow-y-auto">
+      {contentLength && Number(contentLength.value) > FIVE_MEGABYTES ? (
+        <div>This response is larger than 5MB, so it will only be partially displayed</div>
+      ) : (
+        <>
+          <div
+            className="flex items-center py-1 cursor-pointer font-bold"
+            onClick={() => setExpanded(!expanded)}
+          >
+            <TriangleToggle open={expanded} />
+            Response body:
+          </div>
+          {expanded && <div>{validJSON ? <ReactJson src={content} /> : <pre>{content}</pre>}</div>}
```

Comment:
This pre will be showing a bunch of individual base64 pieces, so we probably don't want to render it like this. You can decode each individual part value with https://stackoverflow.com/a/21797381/785065 to get an ArrayBuffer, Then I'd vote to check `Content-Type` to see if it start with `text/` then pass the data to https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder to get the actual text to display. 

If the data _isn't_ `text/*` then we should offer a download link for the URL `const url = URL.createObjectURL(new Blob([arraybuffer], { type: contentType }));`. Note that we should only create the URL temporarily then call `URL.revokeObjectURL` after or else this will leak.

Reply:
I haven't added the download link yet, but otherwise this is done. I think that should be a quick follow-up so we can focus on getting this and request bodies out.